src/features/defi/providers/fox-eth-lp/components/FoxEthLpManager/Deposit/FoxEthLpDeposit.tsx

```diff
@@ -47,20 +47,26 @@ export const FoxEthLpDeposit: React.FC<FoxEthLpDepositProps> = ({
   const translate = useTranslate()
   const { query, history, location } = useBrowserRouter<DefiQueryParams, DefiParams>()
 
-  const accountAddress = useMemo(
-    () => (accountId ? fromAccountId(accountId).account : null),
-    [accountId],
-  )
-
-  const opportunity = useAppSelector(state =>
-    selectFoxEthLpOpportunityByAccountAddress(state, {
-      accountAddress: accountAddress ?? '',
+  const foxEthLpOpportunityFilter = useMemo(
+    () => ({
+      lpId: foxEthLpAssetId as LpId,
+      assetId: foxEthLpAssetId,
     }),
+    [],
+  )
+  const foxEthLpOpportunity = useAppSelector(state =>
+    selectAggregatedEarnUserLpOpportunity(state, foxEthLpOpportunityFilter),
   )
 
-  const asset = useAppSelector(state => selectAssetById(state, opportunity?.assetId ?? ''))
-  const marketData = useAppSelector(state =>
-    selectMarketDataById(state, opportunity?.assetId ?? ''),
+  const foxEthLpAsset = useAppSelector(
+    state =>
+      foxEthLpOpportunity?.underlyingAssetId &&
```

Comment:
let's use constant here

Reply:
Agreed to keep using `underlyingAssetId` from the opportunity for generalization